torch/csrc/autograd/engine.cpp

```diff
@@ -1072,7 +1072,7 @@ size_t Engine::ready_queue_size(const std::shared_ptr<GraphTask>& graph_task, at
 
 // CPU ready queue is per GraphTask, but CUDA device ready queues are shared across all graph tasks
 auto Engine::ready_queue(std::shared_ptr<ReadyQueue> cpu_ready_queue, at::Device device) -> std::shared_ptr<ReadyQueue>{
-  if (device.type() == at::kCPU) {
+  if (device.type() == at::kCPU || device.type() == at::DeviceType::Meta) {
```

Comment:
Wait you send meta Tensors through the autograd????

Reply:
There are some device-generic autograd tests so... yes :)

Reply:
So that means that you can use that to run a full forward/backward without doing any actual computation?

Reply:
in theory! (In practice I don't think we have enough operators implemented yet)

Reply:
Sounds exciting!